capa/features/extractors/vmray/__init__.py

```diff
@@ -0,0 +1,165 @@
+        self.imports: Dict[int, Tuple[str, str]] = {}
+        self.sections: Dict[int, str] = {}
+        self.process_ids: Dict[int, int] = {}
+        self.process_threads: Dict[int, List[int]] = defaultdict(list)
+        self.process_calls: Dict[int, Dict[int, List[FunctionCall]]] = defaultdict(lambda: defaultdict(list))
+        self.base_address: int
+
+        self.sample_file_name: Optional[str] = None
+        self.sample_file_analysis: Optional[File] = None
+        self.sample_file_static_data: Optional[StaticData] = None
+
+        self._find_sample_file()
+
+        if self.sample_file_name is None or self.sample_file_analysis is None:
+            logger.warning("VMRay archive does not contain sample file (file_type: %s)", self.file_type)
+            raise UnsupportedFormatError("VMRay archive does not contain sample file (file_type: %s)", self.file_type)
+
+        if not self.sample_file_static_data:
+            # we see this for text files e.g. JScript file types
+            logger.warning("VMRay archive does not contain static data (file_type: %s)", self.file_type)
+            raise UnsupportedFormatError("VMRay archive does not contain static data (file_type: %s)", self.file_type)
+
+        if not self.sample_file_static_data.pe and not self.sample_file_static_data.elf:
+            logger.warning(
+                "VMRay feature extractor only supports PE and ELF at this time (file_type: %s)", self.file_type
+            )
+            raise UnsupportedFormatError(
+                "VMRay feature extractor only supports PE and ELF at this time(file_type: %s)", self.file_type
+            )
+
+        # VMRay does not store static strings for the sample file so we must use the source file
+        # stored in the archive
+        sample_sha256: str = self.sample_file_analysis.hash_values.sha256.lower()
+        sample_file_path: str = f"internal/static_analyses/{sample_sha256}/objects/files/{sample_sha256}"
+
+        logger.debug("file_type: %s, file_path: %s", self.file_type, sample_file_path)
+
+        self.sample_file_buf: bytes = self.zipfile.read(sample_file_path, pwd=DEFAULT_ARCHIVE_PASSWORD)
+
+        # only compute these if we've found a supported sample file type
+        self._compute_base_address()
+        self._compute_imports()
+        self._compute_exports()
+        self._compute_sections()
+        self._compute_process_ids()
+        self._compute_process_threads()
+        self._compute_process_calls()
+
+    def _find_sample_file(self):
+        for file_name, file_analysis in self.sv2.files.items():
+            if file_analysis.is_sample:
+                # target the sample submitted for analysis
+                self.sample_file_name = file_name
+                self.sample_file_analysis = file_analysis
+
+                if file_analysis.ref_static_data:
+                    self.sample_file_static_data = self.sv2.static_data[file_analysis.ref_static_data.path[1]]
```

Comment:
can you add a comment why it's `path[1]`?

Reply:
https://github.com/mandiant/capa/pull/2208/commits/b967213302a1225c0fbd870d932cc5438ca6b572